### PR TITLE
feat: (TNLT-6012) Trigger article read

### PR DESCRIPTION
**Pull request by aronshamash**
- https://nidigitalsolutions.jira.com/browse/TNLT-6012
- `onArticleRead` will be fired either on scroll of article page or after 6 seconds which ever comes first
- Native will filter calls coming from the in view article.

### CODE REVIEW

File: packages/article-skeleton/__tests__/shared.native.js [hunk 1]
```diff
@@ -186,6 +190,29 @@ export default () => {
...
+    // {
+    //   name: "onArticleRead is called on scroll",
+    //   async test() {
+    //     const onArticleRead = jest.fn();
+
+    //     const testInstance = TestRenderer.create(
+    //       renderArticle({ onArticleRead }),
+    //     );
+
+    //     testInstance.root.findByType(ScrollView).props.onScroll();
```
Comment (aronshamash) [inline packages/article-skeleton/__tests__/shared.native.js:202]: I'm getting this 
```
    TypeError: Cannot read property 'nativeEvent' of undefined

      200 |         );
      201 | 
    > 202 |         testInstance.root.findByType(ScrollView).props.onScroll();
          |                                                        ^
      203 |         expect(onArticleRead).toHaveBeenCalled();
      204 |       },
      205 |     },
```

any ideas ?

File: packages/article-skeleton/__tests__/shared.native.js [hunk 2]
```diff
@@ -186,6 +190,29 @@ export default () => {
...
+    // },
+    {
+      name: "onArticleRead is called after 6 seconds",
+      async test() {
+        const onArticleRead = jest.fn();
+        TestRenderer.create(renderArticle({ onArticleRead }));
+        await delay(6000);
+        expect(onArticleRead).toHaveBeenCalled();
+      },
+      timout: 6000,
```
Comment (aronshamash) [inline packages/article-skeleton/__tests__/shared.native.js:214]: Needed to override the default 5000 jest timeout.

Comment (aronshamash) [reply packages/article-skeleton/__tests__/shared.native.js:214]: I thought better than setting it in jest config

File: packages/article-skeleton/src/article-skeleton.js [hunk 3]
```diff
@@ -48,8 +49,28 @@ const ArticleWithContent = (props) => {
...
+  useEffect(() => {
+    if (!hasBeenRead) {
+      const delay = setTimeout(() => {
+        setArticleRead();
+      }, 6000);
+      return () => clearTimeout(delay);
+    }
+  }, [hasBeenRead]);
+
+  const onScroll = () => {
```
Comment (kamiranoff) [inline packages/article-skeleton/src/article-skeleton.js:70]: Just as a note,
I like to name handler with `handle` (i.e `handleScroll`) It makes it clear it handles the event and leave `on...` for props that are passed through.
basically this naming pattern:
https://jaketrent.com/post/naming-event-handlers-react

File: packages/pages/src/article/article-base.js [hunk 4]
```diff
@@ -71,6 +71,10 @@ const ArticleBase = ({
           error={omitErrors ? null : error}
           interactiveConfig={interactiveConfig}
           isLoading={isLoading || (omitErrors && error)}
+          onArticleRead={(id) => {
+            console.log(`onArticleRead with id: ${id}, fired`);
+            //onArticleRead
```
Comment (kamiranoff) [inline packages/pages/src/article/article-base.js:76]: is this not implemented?

Comment (aronshamash) [reply packages/pages/src/article/article-base.js:76]: no not yet. We have split the feature and the native part has not yet been developed. I can remove the console.log  and comment though so it just does nothing maybe ?

Comment (aronshamash) [reply packages/pages/src/article/article-base.js:76]: it will error until `nativeEvents.onArticleRead` exists.

Comment (kamiranoff) [reply packages/pages/src/article/article-base.js:76]: Ah maybe `// @TODO` but just nitpicking.

Comment (aronshamash) [reply packages/pages/src/article/article-base.js:76]: yes will update

**Review by kamiranoff — APPROVED** [review]

File: packages/article-skeleton/src/article-skeleton.js (hunk 3 above)
Comment (aronshamash) [reply packages/article-skeleton/src/article-skeleton.js]: Changed to : `scrollHandler` now

File: packages/article-skeleton/__tests__/shared.native.js (hunk 2 above)
Comment (aronshamash) [reply packages/article-skeleton/__tests__/shared.native.js]: I used `jest.advanceTimersByTime(6000);`

**Review by c-j-j — APPROVED** [review]

**Pull request merged**